Yosemite/Yosemite/Stores/AnnouncementsStore.swift

```diff
@@ -96,26 +130,80 @@ private extension AnnouncementsStore {
                            iconBase64: $0.iconBase64)
         }
 
-        return StorageAnnouncement(appVersion: announcement.appVersionName,
-                                   features: mappedFeatures,
+        return StorageAnnouncement(appVersionName: announcement.appVersionName,
+                                   minimumAppVersion: announcement.minimumAppVersion,
+                                   maximumAppVersion: announcement.maximumAppVersion,
+                                   appVersionTargets: announcement.appVersionTargets,
+                                   detailsUrl: announcement.detailsUrl,
                                    announcementVersion: announcement.announcementVersion,
+                                   isLocalized: announcement.isLocalized,
+                                   responseLocale: announcement.responseLocale,
+                                   features: mappedFeatures,
                                    displayed: false)
     }
 
-    /// Save the `Announcement` to the appropriate file.
-    func saveAnnouncement(_ announcement: Announcement) throws {
+    /// Map `StorageAnnouncement` to `WordPressKit.Announcement` model
+    func mapStoredAnnouncement(_ storedAnnouncement: StorageAnnouncement) throws -> Announcement {
+        do {
+            let encodedObject = try JSONEncoder().encode(storedAnnouncement)
+            return try JSONDecoder().decode(Announcement.self, from: encodedObject)
+        } catch {
+            throw AnnouncementsStorageError.noAnnouncementSaved
```

Comment:
Shouldn't this be something like `AnnouncementsStorageError.invalidAnnouncement`?

Reply:
Makes total sense. It must have lost context after the refactoring!
Thanks for catching this one too!

Done! ✅